src/globus_sdk/services/auth/client/confidential_client.py

```diff
@@ -294,3 +294,107 @@ def oauth2_token_introspect(
         if include is not None:
             body["include"] = include
         return self.post("/v2/oauth2/token/introspect", data=body, encoding="form")
+
+    # TODO: It is a little strange to place this in this class but Auth treats
+    # client creation different based on how the client was authenticated.
+    # TODO: This is a copy service_client.create_client() minus project_id.
+    def create_child_client(
+        self,
+        name: str,
+        public_client: bool,
+        publicly_visible: bool,
+        *,
+        redirect_uris: t.Iterable[str] | None = None,
+        terms_and_conditions: str | None = None,
+        privacy_policy: str | None = None,
+        required_idp: UUIDLike | None = None,
+        preselect_idp: UUIDLike | None = None,
+        create_params: dict[str, t.Any] | None = None,
```

Comment:
I'm trying to be consistent in new methods in calling any "add-in" dict like this `additional_fields` -- let's follow that in this case.

If you `git grep 'additional_fields'`, you'll see it's all over helpers and methods for various services.
We have some past inconsistency about this, calling it things like `additional_params`, and there are some other one-off names. But my goal is that you can learn `additional_fields` as a name for this when using the SDK and have it always mean the same thing service-to-service.